taxi/forms.py

```diff
@@ -0,0 +1,62 @@
+from django import forms
+from django.contrib.auth import get_user_model
+
+from django.contrib.auth.forms import UserCreationForm, UserChangeForm
+from django.core.exceptions import ValidationError
+
+from .models import Driver, Car
+
+
+class CustomUserCreationForm(UserCreationForm):
+
+    class Meta:
+        model = Driver
+        fields = UserCreationForm.Meta.fields + (
+            "first_name",
+            "last_name",
+            "license_number",
+        )
+
+    def clean_license_number(self):
+        license_number = self.cleaned_data["license_number"]
+        first_three = license_number[:3]
+        last_five = license_number[3:]
+
+        if len(license_number) != 8:
+            raise ValidationError(
+                "Ensure that license consists of 8 characters"
+            )
+
+        elif not first_three.isupper() or not first_three.isalpha():
+            raise ValidationError(
+                "Ensure that first three characters are uppercase letters"
+            )
+
+        elif not last_five.isdigit():
+            raise ValidationError(
+                "Ensure that last five characters are digits"
+            )
+
+        return license_number
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
+
+    class Meta:
+        model = Driver
+        fields = ("license_number", )
+
+    def clean_license_number(self):
+        return CustomUserCreationForm.clean_license_number(self)
```

Comment:
u can just inherit from this class

Reply:
Yes. That was my first idea, but I have a problem. I don't know how to
serve.
When I inherits it from CustomUserCreationForm I have password fields in
license update form an tests doesn't pass. I've tried to delete this fields
password1, password2 but I receive KeyError at /drivers/3/update/license/
'password1'.
What should I do?

Reply:
The best way to make this is to create a mixin )